fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -179,6 +183,48 @@ private IndexingByIndex getIndexerByIndex() {
         if (ex == null) {
             return AsyncUtil.DONE;
         }
+        IndexBuildProto.IndexBuildIndexingStamp conflictingIndexingTypeStamp = common.popConflictingIndexingTypeStamp();
+        if (conflictingIndexingTypeStamp != null) {
+            // Here: an indexing type stamp already exists, with a different type. Few precondition cases are handled
+            if (LOGGER.isInfoEnabled()) {
+                LOGGER.info(KeyValueLogMessage.build("conflicting indexing type stamp",
+                        LogMessageKeys.INDEX_NAME, common.getIndex().getName(),
+                        LogMessageKeys.INDEX_VERSION, common.getIndex().getLastModifiedVersion(),
+                        LogMessageKeys.INDEXER_ID, common.getUuid(),
+                        LogMessageKeys.INDEX_STATE_PRECONDITION, common.getIndexStatePrecondition(),
+                        LogMessageKeys.ACTUAL_TYPE, conflictingIndexingTypeStamp
+                ).toString());
+            }
+            if (common.getIndexStatePrecondition() == IndexStatePrecondition.BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY) {
+                indexer = null;
+                // now, match the policy to the previous run
+                IndexBuildProto.IndexBuildIndexingStamp.Method method = conflictingIndexingTypeStamp.getMethod();
+                if ( method == IndexBuildProto.IndexBuildIndexingStamp.Method.BY_RECORDS) {
+                    fallbackToRecordsScan = true;
+                    return fallbackFunction.get();
+                }
+                if (method == IndexBuildProto.IndexBuildIndexingStamp.Method.BY_INDEX) {
+                    Object sourceIndexSubspaceKey = decodeSubspaceKey(conflictingIndexingTypeStamp.getSourceIndexSubspaceKey());
+                    indexFromIndexPolicy = IndexFromIndexPolicy.newBuilder()
+                            .setSourceIndex(sourceIndexSubspaceKey.toString())
+                            .forbidRecordScan() // no third chance
```

Comment:
What happens if:

1. The source index given has a different subspace key from index subspace key found in the stamp and
2. The new source index can't be used (let's say it's not `READABLE`)

Does it throw an error? Can it ever fall back to a `BY_RECORDS` build (possibly needing to clear out the index data first)?

Reply:
1. At this point - the code will throw an error, letting the caller handle it. 
This makes sense to me when the user chooses `BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE` over
`BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY_REBUILD_IF_POLICY_CHANGED`. In the first case the code is only allowed to continue, in the second case the code may rebuild if needed. 
2. IFI will throw an error, fallback to records-scan, overwrite the type stamp, but then might attempt to continue by the messed up rangeSet. Or in other words - OMG! I shall pay a visit at the planning board. 
I think I should follow the previous rule. I.e. during this scenario, throw an error if `BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE`, rebuild by-records if `BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY_REBUILD_IF_POLICY_CHANGED`. Thoughts?

Reply:
Pushed. 
In the scenario above, if caller asks for `BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY_REBUILD_IF_POLICY_CHANGED`, and the source index subway changes, and fallback is allowed, the fallback will build by records. I think we can live with that...

Reply:
Is there a test for this scenario?